packages/playground/src/portal/twin_view.vue

```diff
@@ -25,15 +40,62 @@
             </v-col>
           </v-row>
         </v-list>
+        <v-card-actions v-if="updateRelay" class="justify-end">
+          <v-btn color="primary" class="custom-button" @click="editTwin">Edit</v-btn>
+        </v-card-actions>
       </v-card>
     </v-container>
   </div>
 </template>
 
 <script lang="ts" setup>
+import { generatePublicKey } from "@threefold/rmb_direct_client";
+import { onMounted, ref } from "vue";
+
 import { useProfileManager } from "../stores";
+import { getGrid } from "../utils/grid";
 
 const profileManager = useProfileManager();
+
+const editingTwin = ref(false);
+const relay = ref(profileManager.profile?.relay || "");
+const updateRelay = ref(false);
+const errorMsg = ref("");
+
+onMounted(validateEdit);
+async function validateEdit() {
+  try {
+    const pk = await generatePublicKey(profileManager.profile!.mnemonic);
+    if (profileManager.profile?.relay !== window.env.RELAY_DOMAIN) {
+      updateRelay.value = true;
+    }
+
+    if (profileManager.profile?.pk !== pk) {
+      UpdateRelay();
+    }
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+function editTwin() {
+  console.log("editing twin");
+  editingTwin.value = true;
+}
+
+async function UpdateRelay() {
+  try {
+    const grid = await getGrid(profileManager.profile!);
+    const newPk = await generatePublicKey(profileManager.profile!.mnemonic);
+    await grid?.twins.update({ relay: relay.value });
+    profileManager.updateRelay(window.env.RELAY_DOMAIN);
```

Comment:
why the profile's relay and the twin's relay are not updated with the same value?